ci: un-skip `pkg/testutils/docker:docker_test` in CI

null

Jira issue: CRDB-16642

Epic CRDB-15060

I tried digging into this one but unfortunately could not find a solution.
Some clues that I have found:
The original error is from https://github.com/cockroachdb/cockroach/blob/6b1c99bb9d907733589c474e44a446ca4dc7cf24/pkg/testutils/docker/single_node_docker_test.go#L521-L529, where we somehow lost the permission to run ./docker-fsnotify . The ./docker-fsnotify binary was built with pkg/testutils/docker/docker-fsnotify . The [error message](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5751145?showRootCauses=true&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildDeploymentsSection=true&logFilter=debug) is
--- FAIL: TestSingleNodeDocker (2.18s)
19:47:59         --- FAIL: TestSingleNodeDocker/single-node-secure-mode (1.93s)
19:47:59             single_node_docker_test.go:221: cannot run fsnotify to listen to init_success:
19:47:59                 res:&docker.execResult{stdOut:"OCI runtime exec failed: exec failed: unable to start container process: exec: \"./docker-fsnotify\": permission denied: unknown\r\n", stdErr:"", exitCode:126}: command "./docker-fsnotify /cockroach init_success 80" exit with code 126:
19:47:59                  {stdOut:OCI runtime exec failed: exec failed: unable to start container process: exec: "./docker-fsnotify": permission denied: unknown
19:47:59      stdErr: exitCode:126}
2. The ./docker-fsnotify binary actually has the Lrwxrwxrwx permission (check Build Log [here](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5516875?buildTab=log&focusLine=4311&linesState=4229.4251) and search in log docker/docker-fsnotify/docker-fsnotify-bin )

Is it ok if I hand this issue back to the dev-inf team?

--- a/build/teamcity/cockroach/ci/tests/docker_image.sh
+++ b/build/teamcity/cockroach/ci/tests/docker_image.sh
@@ -10,9 +10,9 @@ tc_prepare
 tc_start_block "Run docker image tests"
 
 # Skip for now: #82747
-#bazel run \
-#  //pkg/testutils/docker:docker_test \
-#  --config=crosslinux --config=test \
-#  --test_timeout=3000
+bazel run \
+  //pkg/testutils/docker:docker_test \
+  --config=crosslinux --config=test \
+  --test_timeout=3000
 
 tc_end_block "Run docker image tests"

--- a/pkg/testutils/docker/single_node_docker_test.go
+++ b/pkg/testutils/docker/single_node_docker_test.go
@@ -40,8 +40,6 @@ import (
 	"github.com/docker/go-connections/nat"
 )
 
-const fsnotifyBinName = "docker-fsnotify-bin"
-
 // sqlQuery consists of a sql query and the expected result.
 type sqlQuery struct {
 	query          string
@@ -79,7 +77,7 @@ func TestSingleNodeDocker(t *testing.T) {
 		t.Fatal(errors.NewAssertionErrorWithWrappedErrf(err, "cannot get pwd"))
 	}
 
-	fsnotifyPath := filepath.Join(filepath.Dir(filepath.Dir(filepath.Dir(filepath.Dir(filepath.Dir(filepath.Dir(pwd)))))), "docker-fsnotify")
+	fsnotifyBinPath := filepath.Join(pwd, "docker-fsnotify/docker-fsnotify-bin")
 
 	var dockerTests = []singleNodeDockerTest{
 		{
@@ -93,7 +91,7 @@ func TestSingleNodeDocker(t *testing.T) {
 				},
 				volSetting: []string{
 					fmt.Sprintf("%s/testdata/single-node-test/docker-entrypoint-initdb.d/:/docker-entrypoint-initdb.d", pwd),
-					fmt.Sprintf("%s/docker-fsnotify-bin:/cockroach/docker-fsnotify", fsnotifyPath),
+					fmt.Sprintf("%s:/cockroach/docker-fsnotify", fsnotifyBinPath),
 				},
 				cmd: []string{"start-single-node", "--certs-dir=certs"},
 			},
@@ -119,7 +117,7 @@ func TestSingleNodeDocker(t *testing.T) {
 				},
 				volSetting: []string{
 					fmt.Sprintf("%s/testdata/single-node-test/docker-entrypoint-initdb.d/:/docker-entrypoint-initdb.d", pwd),
-					fmt.Sprintf("%s/docker-fsnotify-bin:/cockroach/docker-fsnotify", fsnotifyPath),
+					fmt.Sprintf("%s:/cockroach/docker-fsnotify", fsnotifyBinPath),
 				},
 				cmd: []string{"start-single-node", "--insecure"},
 			},
@@ -146,7 +144,7 @@ func TestSingleNodeDocker(t *testing.T) {
 				},
 				volSetting: []string{
 					fmt.Sprintf("%s/testdata/single-node-test/docker-entrypoint-initdb.d/:/docker-entrypoint-initdb.d", pwd),
-					fmt.Sprintf("%s/docker-fsnotify-bin:/cockroach/docker-fsnotify", fsnotifyPath),
+					fmt.Sprintf("%s:/cockroach/docker-fsnotify", fsnotifyBinPath),
 				},
 				cmd: []string{"start-single-node", "--insecure", "--store=type=mem,size=0.25"},
 			},

